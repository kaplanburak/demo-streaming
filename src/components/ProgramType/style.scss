@import "../../styles/main.scss";

#card {
  background-color: $gray-dark;
  width: 10rem;
  height: 15rem;
  position: relative;
  display: grid;
  place-items: center;
  overflow: hidden;
  cursor: pointer;

  img {
    width: 20rem;
    margin-left: -5rem;
  }

  span {
    color: white;
    font-weight: 500;
    font-size: 2rem;
    position: absolute;
  }
}

#title {
  padding-top: 1rem;
  cursor: pointer;

  span {
    margin: 0;
    font-weight: 500;
  }
}
