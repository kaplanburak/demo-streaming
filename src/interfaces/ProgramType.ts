export interface IProgramTypeProps {
  typeName: string;
}

export interface IProgramTypeState {
  redirect: boolean;
}
