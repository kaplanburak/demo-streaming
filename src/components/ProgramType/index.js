import React from "react";
import { Redirect } from "react-router-dom";
import "./style.scss";
import placeholder from "../../assets/placeholder.png";

class ProgramType extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      redirect: false
    };
  }

  render() {
    const { typeName } = this.props;

    if (this.state.redirect) return <Redirect to={`/${typeName}`} />;

    return (
      <div>
        <div id="card" onClick={() => this.setState({ redirect: true })}>
          <img src={placeholder} alt={typeName} />
          <span>{typeName.toUpperCase()}</span>
        </div>
        <div id="title" onClick={() => this.setState({ redirect: true })}>
          <span>Popular {typeName}</span>
        </div>
      </div>
    );
  }
}

export default ProgramType;
