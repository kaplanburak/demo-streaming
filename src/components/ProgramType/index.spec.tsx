import React from "react";
import { shallow } from "enzyme";
import ProgramType from "./index";

describe("ProgramType component", () => {
  const programType = shallow(<ProgramType typeName={"Programs"} />);

  it("should render", () => {
    expect(programType).toMatchSnapshot();
  });

  it("should show title on card", () => {
    expect(
      programType
        .find("#card")
        .childAt(1)
        .text()
    ).toEqual("PROGRAMS");
  });

  it("should show title below the card", () => {
    expect(
      programType
        .find("#title")
        .childAt(0)
        .text()
    ).toEqual("Popular Programs");
  });
});
