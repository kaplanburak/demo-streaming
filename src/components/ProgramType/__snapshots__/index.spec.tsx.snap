// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramType component should render 1`] = `ShallowWrapper {}`;
